### PR TITLE
Fix barelf URL

### LINKED ISSUE #2956: The link in the "451 Bare line-feed; see http://haraka.github.com/barelf.html" error message is broken
**Describe the bug**
When sending dodgy data to Haraka, a "451 Bare line-feed" error can be triggered - however the link for further information is broken and returns a 404 page

**Expected behavior**
The link to point to a live page (presumably https://haraka.github.io/barelf/)

**Observed behavior**
Error message contains broken link
```
451 Bare line-feed; see http://haraka.github.com/barelf.html
```

**Steps To Reproduce**
Send dodgy data to Haraka

**System Info:**

Haraka | Haraka.js — Version: 2.8.27
 --- | :---
Node | v10.19.0
OS | Linux haraka 5.4.0-54-generic #60-Ubuntu SMP Fri Nov 6 10:37:59 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
openssl | OpenSSL 1.1.1f  31 Mar 2020

**Additional context**
Appears to be defined in connection.js
https://github.com/haraka/Haraka/blob/e3bafbbe0b66b5eea5c5753808fef3473b8f2dfd/connection.js#L1591

### PR DESCRIPTION
**Pull request by baudehlo**
Fixes #2956

Changes proposed in this pull request:
- 
- 

Checklist:
- [ ] docs updated
- [ ] tests updated
- [ ] [Changes](https://github.com/haraka/Haraka/blob/master/Changes.md) updated

### CODE REVIEW

**Comment by sent** [comment]
Hello @baudehlo,

Actually, it's not a problem with that exact URL, however with that domain https://github.blog/changelog/2021-01-29-github-pages-will-stop-redirecting-pages-sites-from-github-com-after-april-15-2021/ 

Redirection works fine: https://haraka.github.io/barelf.html

Thanks!

File: connection.js [hunk 1]
```diff
@@ -1588,7 +1588,7 @@ class Connection {
             line[0] === 0x2e &&
             line[1] === 0x0a) {
             this.lognotice('Client sent bare line-feed - .\\n rather than .\\r\\n');
-            this.respond(451, "Bare line-feed; see http://haraka.github.com/barelf.html", () => {
+            this.respond(451, "Bare line-feed; see http://haraka.github.com/barelf/", () => {
```
Comment (msimerson) [inline connection.js:1591]:
```suggestion
            this.respond(451, "Bare line-feed; see http://haraka.github.io/barelf/", () => {
```

**Review by msimerson — CHANGES_REQUESTED** [review]

**Comment by christianflintrup** [comment]
Seems the new URL (`http://haraka.github.io/barelf/`) also returns a 404 now.

**Comment by sent** [comment]
> Seems the new URL (`http://haraka.github.io/barelf/`) also returns a 404 now.

Fixed in https://github.com/haraka/haraka.github.io/pull/43

**Pull request merged**